src/jswrap_object.c

```diff
@@ -222,8 +222,20 @@ void jswrap_object_keys_or_property_names_cb(
 
     while (symbols) {
       unsigned int i;
-      for (i=0;i<symbols->symbolCount;i++) {
-        JsVar *name = jsvNewFromString(&symbols->symbolChars[symbols->symbols[i].strOffset]);
+      unsigned char symbolCount = READ_FLASH_UINT8(&symbols->symbolCount);
+      unsigned short strOffset = READ_FLASH_UINT16(&symbols->symbols[i].strOffset);
+      for (i=0;i<symbolCount;i++) {
+#ifndef USE_FLASH_MEMORY
+        JsVar *name = jsvNewFromString(&symbols->symbolChars[strOffset]);
+#else
+        // On the esp8266 the string is in flash, so we have to copy it to RAM first
+        // We can't use flash_strncpy here because it assumes that strings start on a word
```

Comment:
Shame... just looked and there is a 'rev' instruction in thumb code for swapping endianness, so it would still be pretty quick to do. Not sure if it's worth it though :)